content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/retry_state.rb

```diff
@@ -0,0 +1,72 @@
+#
+# Description: This class updates the provisioning status and retries
+# from the desired step.
+# Required inputs: status
+# Optional inputs: max_retries, error_to_catch, restart_from_state
+#
+module ManageIQ
+  module Automate
+    module Service
+      module Generic
+        module StateMachines
+          module VMProvision_VM
+            class RetryState
+              DEFAULT_MAX_RETRIES = 3
+              DEFAULT_ERROR_TO_CATCH = "An error occurred while provisioning Instance".freeze
+              DEFAULT_RESTART_FROM_STATE = "Placement".freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+                @prov = @handle.root['miq_provision']
+                @max_retries = @handle.inputs['max_retries'] || DEFAULT_MAX_RETRIES
+                @restart_from_state = @handle.inputs['restart_from_state'] || DEFAULT_RESTART_FROM_STATE
+                @error_to_catch = @handle.inputs['error_to_catch'] || DEFAULT_ERROR_TO_CATCH
+                @retry_number = if @handle.state_var_exist?(:state_retries)
+                                  @handle.get_state_var(:state_retries)
+                                else
+                                  0
+                                end
+              end
+
+              def main
+                @handle.log("info", "Starting retry_state")
+                retry_state
```

Comment:
My intention is just to affect Openstack provisioning. I might have made a mistake with my scoping.